### PR TITLE
Avoid pre-fetching for unbounded minimum versions

### PR DESCRIPTION
**Pull request by charliermarsh**
## Summary

I think we should be able to model PubGrub such that this isn't necessary (at least for the case described in the issue), but for now, let's just avoid attempting to build very old distributions in prefetching.

Closes https://github.com/astral-sh/uv/issues/4136.

### CODE REVIEW

File: crates/uv-resolver/src/resolver/mod.rs [hunk 1]
```diff
@@ -1402,6 +1403,18 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                     return Ok(None);
                 };
 
+                // Avoid prefetching source distributions with unbounded lower-bound ranges.
```
Comment (zanieb) [inline crates/uv-resolver/src/resolver/mod.rs:1406]: Should we say why here?

**Review by zanieb — APPROVED** [review]

File: crates/uv-resolver/src/resolver/mod.rs (hunk 1 above)
Comment (zanieb) [reply crates/uv-resolver/src/resolver/mod.rs:1406]: I guess it's in the other comment "could be expensive?"

File: crates/uv-resolver/src/resolver/mod.rs [hunk 2]
```diff
@@ -1402,6 +1403,20 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                     return Ok(None);
                 };
 
+                // Avoid prefetching source distributions with unbounded lower-bound ranges. This
+                // often leads to failed attempts to build legacy versions of packages that are
+                // incompatible with modern build tools.
```
Comment (zanieb) [inline crates/uv-resolver/src/resolver/mod.rs:1408]: Love it :)

Comment (potiuk) [reply crates/uv-resolver/src/resolver/mod.rs:1408]: Nice, simple solution to a simple problem. Let met confirm it works :)

**Pull request merged**